### ISSUE TITLE
Link annotations

### ISSUE THREAD

**Issue by matthiasbeyer** [matthiasbeyer/imag#936]
Continue the work started in https://github.com/matthiasbeyer/imag/pull/904.

We need

- [x] Tests for the annotation functionality
- [ ] Full CRUD support

**Comment by matthiasbeyer** [comment]
This can be done on the "lib"-level plus on a rather high level (`imag-link`) after #947 is merged.

Sidenote for @irobert91: I wouldn't include this in #947 but a follow-up PR.
